core/src/main/java/com/linecorp/armeria/server/ServiceConfigBuilder.java

```diff
@@ -302,18 +305,46 @@ ServiceConfig build(ServiceNaming defaultServiceNaming,
                                                                      unhandledExceptionsReporter);
         }
 
+        final boolean webSocket = service.as(WebSocketService.class) != null;
+        final long requestTimeoutMillis;
+        if (this.requestTimeoutMillis != null) {
+            requestTimeoutMillis = this.requestTimeoutMillis;
+        } else if (!webSocket || defaultRequestTimeoutMillis != Flags.defaultRequestTimeoutMillis()) {
+            requestTimeoutMillis = defaultRequestTimeoutMillis;
+        } else {
+            requestTimeoutMillis = WebSocketServiceUtil.DEFAULT_REQUEST_TIMEOUT_MILLIS;
+        }
```

Comment:
I don't think I understood this condition 😅 Couldn't we just do the following since `defaultRequestTimeoutMillis` will be set to `Flags.defaultRequestTimeoutMillis()` for `virtualHostTemplate` by default anyways?

```suggestion
        } else if (!webSocket) {
            requestTimeoutMillis = defaultRequestTimeoutMillis;
        } else {
            requestTimeoutMillis = WebSocketServiceUtil.DEFAULT_REQUEST_TIMEOUT_MILLIS;
        }
```

Reply:
The value set through `ServerBuilder.requestTimeoutMillis(value)` is the `defaultRequestTimeoutMillis`. So it could be different from `Flags.defaultRequestTimeoutMillis()`. 😉